eng/slngen.targets

```diff
@@ -4,11 +4,8 @@
     <SlnGenSolutionFolder Condition="'$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true'">tests</SlnGenSolutionFolder>
     <SlnGenSolutionFolder Condition="'$(SlnGenSolutionFolder)' == ''">src</SlnGenSolutionFolder>
 
-    <!-- Don't include shared framework projects referenced via a named reference except the leaf's project. -->
-    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' != '$(SlnGenMainProject)'">false</IncludeInSolutionFile>
-
-    <!-- Include CompilerServices.Unsafe as it's a transitive test dependency. -->
-    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' == 'System.Runtime.CompilerServices.Unsafe'">true</IncludeInSolutionFile>
+    <!-- Don't include shared framework projects except the leaf's project and OOBs. -->
+    <IncludeInSolutionFile Condition="'$(IsNETCoreAppRef)' == 'true' and '$(MSBuildProjectName)' != '$(SlnGenMainProject)' and '$(IsNETCoreAppPackage)' != 'true'">false</IncludeInSolutionFile>
```

Comment:
After doing this I realize that another condition that might work is `'$(TargetFrameworks)' != '$(NetCoreAppCurrent)'` since this will also catch references that didn't build in the up-front build pass.  Let me know what makes more sense to use.

Reply:
Yes, I would prefer that over the static list (which we likely need anyway for another change but probably not here) as the existence of another TargetFramework than `$(NetCoreAppCurrent)*` should indicate that the reference project should be present in the solution file.